internal/graphicsdriver/metal/mtl/mtl_darwin.go

```diff
@@ -457,134 +461,236 @@ type Device struct {
 }
 
 // MakeTexture creates a texture object with privately owned storage
 // that contains texture state.
 //
 // Reference: https://developer.apple.com/documentation/metal/mtldevice/1433425-maketexture.
 func (d Device) MakeTexture(td TextureDescriptor) Texture {
-	descriptor := C.struct_TextureDescriptor{
-		TextureType: C.uint16_t(td.TextureType),
-		PixelFormat: C.uint16_t(td.PixelFormat),
-		Width:       C.uint_t(td.Width),
-		Height:      C.uint_t(td.Height),
-		StorageMode: C.uint8_t(td.StorageMode),
-		Usage:       C.uint8_t(td.Usage),
-	}
+	textureDescriptor := objc.ID(class_MTLTextureDescriptor).Send(sel_new)
+	textureDescriptor.Send(sel_setTextureType, uintptr(td.TextureType))
+	textureDescriptor.Send(sel_setPixelFormat, uintptr(td.PixelFormat))
+	textureDescriptor.Send(sel_setWidth, uintptr(td.Width))
+	textureDescriptor.Send(sel_setHeight, uintptr(td.Height))
+	textureDescriptor.Send(sel_setStorageMode, uintptr(td.StorageMode))
+	textureDescriptor.Send(sel_setUsage, uintptr(td.Usage))
+	texture := d.device.Send(sel_newTextureWithDescriptor, textureDescriptor)
+	textureDescriptor.Send(sel_release)
 	return Texture{
-		texture: C.Device_MakeTexture(d.device, descriptor),
+		texture: texture,
 	}
 }
 
 // MakeDepthStencilState creates a new object that contains depth and stencil test state.
 //
 // Reference: https://developer.apple.com/documentation/metal/mtldevice/1433412-makedepthstencilstate
 func (d Device) MakeDepthStencilState(dsd DepthStencilDescriptor) DepthStencilState {
-	descriptor := C.struct_DepthStencilDescriptor{
-		BackFaceStencilStencilFailureOperation:    C.uint8_t(dsd.BackFaceStencil.StencilFailureOperation),
-		BackFaceStencilDepthFailureOperation:      C.uint8_t(dsd.BackFaceStencil.DepthFailureOperation),
-		BackFaceStencilDepthStencilPassOperation:  C.uint8_t(dsd.BackFaceStencil.DepthStencilPassOperation),
-		BackFaceStencilStencilCompareFunction:     C.uint8_t(dsd.BackFaceStencil.StencilCompareFunction),
-		FrontFaceStencilStencilFailureOperation:   C.uint8_t(dsd.FrontFaceStencil.StencilFailureOperation),
-		FrontFaceStencilDepthFailureOperation:     C.uint8_t(dsd.FrontFaceStencil.DepthFailureOperation),
-		FrontFaceStencilDepthStencilPassOperation: C.uint8_t(dsd.FrontFaceStencil.DepthStencilPassOperation),
-		FrontFaceStencilStencilCompareFunction:    C.uint8_t(dsd.FrontFaceStencil.StencilCompareFunction),
-	}
+	depthStencilDescriptor := objc.ID(class_MTLDepthStencilDescriptor).Send(sel_new)
+	backFaceStencil := depthStencilDescriptor.Send(sel_backFaceStencil)
+	backFaceStencil.Send(sel_setStencilFailureOperation, uintptr(dsd.BackFaceStencil.StencilFailureOperation))
+	backFaceStencil.Send(sel_setDepthFailureOperation, uintptr(dsd.BackFaceStencil.DepthFailureOperation))
+	backFaceStencil.Send(sel_setDepthStencilPassOperation, uintptr(dsd.BackFaceStencil.DepthStencilPassOperation))
+	backFaceStencil.Send(sel_setStencilCompareFunction, uintptr(dsd.BackFaceStencil.StencilCompareFunction)) //TODO
```

Comment:
What is this TODO? @TotallyGamerJet

Reply:
It can be removed. It was needed when I was working on it and just never got deleted.

Reply:
Sure!